internal/datastore/proxy/observable_test.go

```diff
@@ -0,0 +1,28 @@
+package proxy
+
+import (
+	"testing"
+	"time"
+
+	"github.com/authzed/spicedb/internal/datastore/memdb"
+	"github.com/authzed/spicedb/pkg/datastore"
+	"github.com/authzed/spicedb/pkg/datastore/test"
+)
+
+type observableTest struct{}
+
+func (obs observableTest) New(revisionQuantization, _, gcWindow time.Duration, watchBufferLength uint16) (datastore.Datastore, error) {
```

Comment:
Adding this makes it so we test all of the Datastore tests with the Observable proxy, and more specifically the `DeleteWithLimitTest` which verifies the fix in this PR